onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -288,11 +288,9 @@ const char* GetDeviceName(const OrtDevice& device) {
     case OrtDevice::CPU:
       return CPU;
     case OrtDevice::GPU:
-#ifdef USE_DML
-      return DML;
-#else
       return CUDA;
-#endif
+    case OrtDevice::DML:
+      return DML;
```

Comment:
I see parts of the DML EP use `OrtDevice::GPU`, and so I wonder the impact of splitting this out; but it appears this change would only impact Python callers at most (not existing C++ callers).

```
    BucketizedBufferAllocator::BucketizedBufferAllocator(
        ID3D12Device* device,
        ExecutionContext* context,
        const D3D12_HEAP_PROPERTIES& heapProps,
        D3D12_HEAP_FLAGS heapFlags,
        D3D12_RESOURCE_FLAGS resourceFlags,
        D3D12_RESOURCE_STATES initialState,
        std::unique_ptr<DmlSubAllocator>&& subAllocator
        )
        : onnxruntime::IAllocator(
            OrtMemoryInfo(
                "DML",
                OrtAllocatorType::OrtDeviceAllocator,
                OrtDevice(OrtDevice::GPU, OrtDevice::MemType::DEFAULT, 0) <<<<<<<<<<<<<<<<<
            )
        ),
```

```
    ExecutionProvider::ExecutionProvider(
        IDMLDevice* dmlDevice,
        Dml::ExecutionContext* executionContext,
        bool enableMetacommands,
        bool enableGraphCapture,
        bool enableSyncSpinning,
        bool disableMemoryArena) :
            IExecutionProvider(onnxruntime::kDmlExecutionProvider, OrtDevice(OrtDevice::GPU, OrtDevice::MemType::DEFAULT, 0))  <<<<<<<<<<<<<<<<<
```

Reply:
Thanks for pointing this out. Fixed in the next commit. It's good to keep things consistent.

Reply:
I just checked my little DML sample (https://github.com/fdwr/OnnxRuntimeDirectMLEPSample), and I don't explicitly refer to the `OrtMemoryInfoDeviceType` anywhere anyway, just memoryInformationName ("DML", "Cpu"), `OrtAllocatorType` (`OrtDeviceAllocator`), and `OrtMemType` (`OrtMemTypeDefault`).

https://github.com/microsoft/onnxruntime/blob/b4afc6266f7ff20e7b79eaea7fa62f3e30b7474f/include/onnxruntime/core/session/onnxruntime_c_api.h#L393

It's interesting the difference between this enum set and the public enum:

```
typedef enum OrtMemoryInfoDeviceType {
  OrtMemoryInfoDeviceType_CPU = 0,
  OrtMemoryInfoDeviceType_GPU = 1,
  OrtMemoryInfoDeviceType_FPGA = 2
} OrtMemoryInfoDeviceType;
```